### PR TITLE
Issue 8 updatable staking

### PR DESCRIPTION
**Pull request by mfahampshire**

### CODE REVIEW

**Comment by mfahampshire** [comment]
Procedure for initializing `StakingRewards.sol` is now: 
- Deploy contract 
- Fund contract with reward amount 
- call `notifyRewardAmount()` with funded amount 

Procedure for topping up contract / extending staking period is now: 
- Fund contract with extra reward amount 
- call `notifyRewardAmount()` with extra reward amount

File: contracts/incentives/StakingRewards.sol [hunk 1]
```diff
@@ -67,10 +68,7 @@ contract StakingRewards is Ownable, ReentrancyGuard {
         starttime = _starttime;
         DURATION = (_duration * 24 hours);
 
-        require(_initreward == IERC20(rewardToken).balanceOf(address(this)),
-                "StakingRewards: wrong reward amount supplied");
-
-        _notifyRewardAmount(_initreward);
+        rewardDistribution = msg.sender;
```
Comment (kinspotting) [inline contracts/incentives/StakingRewards.sol:71]: In this case `msg.sender` would `FarmFactory` contract what seems to be not optimal, I guess we should pass `avatar` address to the contract so the `dao` has the `rewardDistribution` rights

File: test/staking-simulation-1-month-happypath.js [hunk 2]
```diff
@@ -194,10 +195,10 @@ contract('Staking: 1 month happypath', (accounts) => {
                 it('Contract bPRIME balance == 0', async () => {
                     expect((await setup.balancer.pool.balanceOf(setup.incentives.stakingRewards.address)).toString()).to.equal('0'); // all stake removed
                 });
-                it('reduction in stakingRewards prime balance == ~total reward payout amount', async () => {
+                it('reduction in stakingRewards prime balance > 99.999% of total reward amount', async () => {
                     let remainingPrimeBalance = BigNumber(await setup.tokens.primeToken.balanceOf(setup.incentives.stakingRewards.address));
 
-                    console.log('            remainingPrimeBalance: ' + remainingPrimeBalance.toString() + '/92500000000000002949500');
+                    // console.log('            remainingPrimeBalance: ' + remainingPrimeBalance.toString() + '/' + _initreward);
```
Comment (kinspotting) [inline test/staking-simulation-1-month-happypath.js:201]: please delete comment

File: test/staking-simulation-1-week-happypath.js [hunk 3]
```diff
@@ -193,15 +194,15 @@ contract('Staking: 1 week happypath', (accounts) => {
...
+                it('reduction in stakingRewards prime balance > 99.999% of total reward amount', async () => {
                     let remainingPrimeBalance = BigNumber(await setup.tokens.primeToken.balanceOf(setup.incentives.stakingRewards.address));
 
                     /*
                     * Remaining token balance is: ~190000 if everyone calls exit() on day 7 (which considering we're using an 18 decimal token is fractional)
                     *                             ~13213979828042328659724 if everyone calls exit() on day 6
                     *                             ~26428265542328043366944 if everyone calls exit() on day 5
                     */
-                    console.log('            remainingPrimeBalance: ' + remainingPrimeBalance.toString() + '/92500000000000002949500');
+                    // console.log('            remainingPrimeBalance: ' + remainingPrimeBalance.toString() + '/' + _initreward);
```
Comment (kinspotting) [inline test/staking-simulation-1-week-happypath.js:205]: same

**Review by kinspotting — CHANGES_REQUESTED** [review]
Needs changes

File: contracts/incentives/StakingRewards.sol (hunk 1 above)
Comment (mfahampshire) [reply contracts/incentives/StakingRewards.sol:71]: Good point - will rework this :)

**Review by kinspotting — APPROVED** [review]
lgtm

**Pull request merged**